### PR TITLE
vec_shuffle: Add a keccak testsuite

### LINKED ISSUE #121: `vec_shuffle` test vectors are outdated
## Description

The YAML test vectors for [utils/vec_shuffle](https://github.com/sigp/lighthouse/tree/master/beacon_chain/utils/vec_shuffle/src/specs) use blake2 instead of keccak.

## Steps to resolve

- Reach out on the gitter.im/ethereum/sharding gitter to find new test vectors.
- Implement test vectors.

If our code does not pass the tests, please raise a new issue.

### PR DESCRIPTION
**Pull request by drozdziak1**
## Issue Addressed
Closes #121.

## Proposed Changes
I asked on `ethereum/sharding` and Mamy recommended
https://github.com/status-im/eth2-testgen/, so I just grabbed the yaml file from [here](https://github.com/status-im/eth2-testgen/blob/master/test_vectors/test_vector_shuffling.yml)

## Additional Info
The file is CC0-licensed. I believe this is blocked by #109  before I even know how to hook this up to the unit test.

### CODE REVIEW

**Comment by paulhauner** [comment]
Hey, thanks for jumping on this one!

I can see these tests are targeted at a higher-level than the present tests we have. Our `vec_shuffle` crate is only concerned with shuffling a `Vec<T>`. These tests are concerned with assigning validators to shards, as well as doing a core shuffle.

Put differently, our `vec_shuffle` crate mirrors the `shuffle()` function in the spec, whilst the status-im tests target the `get_new_shuffling()` function. [Reference](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#beacon-chain-state-transition-function).

We haven't implemented the higher-level `get_new_shuffling` fn yet, we're waiting on the spec to settle, so these tests aren't immediately useful. Sorry, I didn't realize Mamy's tests would be at this higher level.

Moving forward, the YAML vectors we have are generated from [this](https://github.com/sigp/shuffling_sandbox/blob/0cf99ae8ccdf1f6370e27c727317707f9ab8e20b/sandbox.py#L123-L165) Python code. To generate new, more relevant vectors we'd need to update the [shuffle function](https://github.com/sigp/shuffling_sandbox/blob/master/src/shufflers/v2_1_spec.py) in that repo to use keccak and then re-generate the YAML.

Is this something you'd be interested in doing?

**Comment by drozdziak1** [comment]
Updated the testsuite generator in sigp/shuffling_sandbox#2, generated a file with my changed script pushed it to my feature branch. If that PR looks right then this PR is probably only blocked by #109 and some small bugs that will appear when trying to make that test happy

**Comment by drozdziak1** [comment]
#109 is done and works with what I generated. Not WIP anymore to me

**Comment by paulhauner** [comment]
Got it, thanks. Will jump onto this in the coming days :)

**Comment by drozdziak1** [comment]
Rebased on top of #122 just to be sure

**Comment by drozdziak1** [comment]
Rebased #125, #126 and #114

**Comment by paulhauner** [comment]
Hey, sorry about the delay but I'm onto this now/

File: beacon_chain/utils/vec_shuffle/src/lib.rs [hunk 1]
```diff
@@ -75,7 +78,9 @@ mod tests {
 
             let mut s = shuffle(&seed, input).unwrap();
 
+            println!("Different: {}", s != output);
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply beacon_chain/utils/vec_shuffle/src/lib.rs]: I did not leave this in intentionally, just debug noise, let me get that

**Comment by drozdziak1** [comment]
Removed the printlns, rebased #124 and #128

**Comment by paulhauner** [comment]
TY, LGTM!

**Pull request merged**